File: frontend/src/types/invitation/InvitationDTO.ts
import { UserRole } from "../UserRole";
import { InvitationStatus } from "./InvitationStatus";

export interface InvitationDTO {
    id: number; 
    email: string; 
    role: UserRole; 
    used: boolean;
    expiresAt: string; 
    status: InvitationStatus;
  }
  

File: backend/src/main/java/com/imbus/knowledge/User_Management/dto/InvitationDTO.java
package com.imbus.knowledge.User_Management.dto;

import com.imbus.knowledge.User_Management.entities.InvitationStatus;
import com.imbus.knowledge.User_Management.entities.UserRole;

import java.time.LocalDateTime;

public record InvitationDTO(
        Long id,
        String email,
        UserRole role,
        boolean used,
        LocalDateTime expiresAt,
        InvitationStatus status
) {}
